l2-contracts/src/updateL2ERC20Metadata.ts

```diff
@@ -0,0 +1,190 @@
+  ]);
+}
+
+async function getReinitializeTokenTxInfo(
+  deployer: Deployer,
+  refundRecipient: string,
+  gasPrice: BigNumber,
+  tokenAddress: string,
+  newName: string,
+  newSymbol: string,
+  ignoreNameGetter: boolean,
+  ignoreSymbolGetter: boolean,
+  ignoreDecimalsGetter: boolean,
+  version: ethers.BigNumberish
+) {
+  const l2Calldata = await getReinitializeTokenCalldata(
+    tokenAddress,
+    newName,
+    newSymbol,
+    ignoreNameGetter,
+    ignoreSymbolGetter,
+    ignoreDecimalsGetter,
+    version
+  );
+  return await getL1TxInfo(
+    deployer,
+    tokenAddress,
+    l2Calldata,
+    refundRecipient,
+    gasPrice,
+    priorityTxMaxGasLimit,
+    provider
+  );
+}
+
+async function main() {
+  const program = new Command();
+
+  program.version("0.1.0").name("update-l2-erc20-metadata");
+
+  program
+    .option("--token-address <upgrades-info>")
+    .option("--gas-price <gas-price>")
+    .option("--deployer-private-key <deployer-private-key>")
+    .option("--refund-recipient <refund-recipient>")
+    .option("--new-name <new-name>")
+    .option("--new-symbol <new-symbol>")
+    .option("--ignore-name-getter")
+    .option("--ignore-symbol-getter")
+    .option("--ignore-decimals-getter")
+    .option("--reinitialization-version <version>")
+    .action(async (cmd) => {
+      const gasPrice = cmd.gasPrice
+        ? ethers.utils.parseUnits(cmd.gasPrice, "gwei")
+        : (await provider.getGasPrice()).mul(3).div(2);
+      const deployWallet = cmd.deployerPrivateKey
+        ? new Wallet(cmd.deployerPrivateKey, provider)
+        : Wallet.fromMnemonic(
+            process.env.MNEMONIC ? process.env.MNEMONIC : ethTestConfig.mnemonic,
+            "m/44'/60'/0'/0/1"
+          ).connect(provider);
+      const deployer = new Deployer({ deployWallet });
+      const refundRecipient = cmd.refundRecipient ? cmd.refundRecipient : deployWallet.address;
+      console.log("Gas price: ", ethers.utils.formatUnits(gasPrice, "gwei"));
+      console.log(
+        "IMPORTANT: gasPrice that you provide in the transaction should be <= to the one provided to this tool."
+      );
+
+      console.log("Refund recipient: ", refundRecipient);
+
+      const tokenAddress = cmd.tokenAddress;
+      const newName = cmd.newName;
+      const newSymbol = cmd.newSymbol;
+      const ignoreNameGetter = cmd.ignoreNameGetter ?? false;
+      const ignoreSymbolGetter = cmd.ignoreSymbolGetter ?? false;
+      const ignoreDecimalsGetter = cmd.ignoreDecimalsGetter ?? false;
+
+      let version;
+      if (cmd.reinitializationVersion === undefined) {
+        const provider = new Provider(process.env.API_WEB3_JSON_RPC_HTTP_URL);
+        const initializableStorageValue = BigNumber.from(
+          await provider.getStorageAt(tokenAddress, INITIALIZABLE_STORAGE_SLOT)
+        );
+
+        // currently it's saved in the first `uint64` struct field
+        const initializedValue = initializableStorageValue.mod(BigNumber.from(2).pow(64));
```

Comment:
mos 2^8 probably, since it is `uint8` value

Reply:
indeed, and also fixed some more issues, since I was looking into the wrong openzeppelin version. Thanks